airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -46,74 +46,88 @@ class DeclarativeOauth2Authenticator(AbstractOauth2Authenticator, DeclarativeAut
     refresh_token: Optional[Union[InterpolatedString, str]] = None
     scopes: Optional[List[str]] = None
     token_expiry_date: Optional[Union[InterpolatedString, str]] = None
-    _token_expiry_date: pendulum.DateTime = field(init=False, repr=False, default=None)
-    token_expiry_date_format: str = None
+    _token_expiry_date: Optional[pendulum.DateTime] = field(init=False, repr=False, default=None)
+    token_expiry_date_format: Optional[str] = None
     token_expiry_is_time_of_expiration: bool = False
     access_token_name: Union[InterpolatedString, str] = "access_token"
     expires_in_name: Union[InterpolatedString, str] = "expires_in"
     refresh_request_body: Optional[Mapping[str, Any]] = None
     grant_type: Union[InterpolatedString, str] = "refresh_token"
     message_repository: MessageRepository = NoopMessageRepository()
 
-    def __post_init__(self, parameters: Mapping[str, Any]):
-        self.token_refresh_endpoint = InterpolatedString.create(self.token_refresh_endpoint, parameters=parameters)
-        self.client_id = InterpolatedString.create(self.client_id, parameters=parameters)
-        self.client_secret = InterpolatedString.create(self.client_secret, parameters=parameters)
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        super().__init__()
+        self._token_refresh_endpoint = InterpolatedString.create(self.token_refresh_endpoint, parameters=parameters)
```

Comment:
was mypy complaining about the non-private fields for refresh, client id and client secret or some name collision? I noticed you made this change, but not to the `access_token_name`, `expires_in_name`, and `grant_type` which follow the same naming convention, so curious why this was needed for only some?

Reply:
yes. taking the refresh token as an example, if using the public variable, mypy complains on [line 82](https://github.com/airbytehq/airbyte/pull/32967/files#diff-7f106d44561d0c12051aba82bbd28dc618fe945d50efbfeb4847608588d6664bR82) because the value is expected to be an interpolated string, not an Union[str, InterpolatedString]